### PR TITLE
rrdtool: Use SVG images to make reports nicely scalable

### PR DESCRIPTION
**Pull request by ffloimair**
Using SVG not only improves the readability of the graphs but
also makes them scalable on all sorts of resolutions or zoom
levels (e.g. on HiDpi/Retina screens)

Additionally make graphs slightly bigger (+50%) by default.

### CODE REVIEW

**Comment by votdev** [comment]
Please adapt your PR according to https://openmediavault.readthedocs.io/en/latest/development/contribute.html.

**Comment by ffloimair** [comment]
OK. I updated the initial commit with the Signed-off-by line and added a line to the changelog.

**Comment by ffloimair** [comment]
Changed as requested.

**Comment by votdev** [comment]
I get segfaults when creating the graphs:
```
omv-mkrrdgraph 
Segmentation fault
Segmentation fault
Segmentation fault
Segmentation fault
Segmentation fault
Segmentation fault
Segmentation fault
Segmentation fault
Segmentation fault
Segmentation fault
Segmentation fault
Segmentation fault
Segmentation fault
Segmentation fault
Segmentation fault
```

**Comment by ffloimair** [comment]
Interesting. Works for me. I did however test the change with OMV 4 as the underlying version.
I will create a new VM, install all the latest stuff and get back to you.

Is there a proposed way of setting up a test-system for Version 5?

**Comment by votdev** [comment]
You need to change the env variable OMV_COLLECTD_RRDTOOL_GRAPH_IMGFORMAT from PNG to SVG in usr/sbin/omv-mkrrdgraph.

**Comment by votdev** [comment]
The JS code still loads the PNG images:
```
diff --git a/deb/openmediavault/usr/sbin/omv-mkrrdgraph b/deb/openmediavault/usr/sbin/omv-mkrrdgraph
index 41ad53a16..af84797b6 100755
--- a/deb/openmediavault/usr/sbin/omv-mkrrdgraph
+++ b/deb/openmediavault/usr/sbin/omv-mkrrdgraph
@@ -48,7 +48,7 @@ def main():
                        '--daemon', 'unix:{}'.format(openmediavault.getenv(
                                'OMV_RRDCACHED_SOCKETFILE', '/var/run/rrdcached.sock')),
                        '--imgformat', openmediavault.getenv(
-                               'OMV_COLLECTD_RRDTOOL_GRAPH_IMGFORMAT', 'PNG'),
+                               'OMV_COLLECTD_RRDTOOL_GRAPH_IMGFORMAT', 'SVG'),
                        '--width', openmediavault.getenv(
                                'OMV_COLLECTD_RRDTOOL_GRAPH_WIDTH', '400'),
                        '--height', openmediavault.getenv(
diff --git a/deb/openmediavault/var/www/openmediavault/js/omv/workspace/panel/RrdGraph.js b/deb/openmediavault/var/www/openmediavault/js/omv/workspace/panel/RrdGraph.js
index 3e01ed8d5..3bea740df 100644
--- a/deb/openmediavault/var/www/openmediavault/js/omv/workspace/panel/RrdGraph.js
+++ b/deb/openmediavault/var/www/openmediavault/js/omv/workspace/panel/RrdGraph.js
@@ -58,11 +58,11 @@ Ext.define("OMV.workspace.panel.RrdGraph", {
        getTemplate: function() {
                return Ext.create("Ext.XTemplate",
                  '<div class="x-panel-rrdgraph">',
-                 '  <img src="rrd.php?name={name}-hour.png&time={time}" alt="RRD graph - by hour"/><br/>',
-                 '  <img src="rrd.php?name={name}-day.png&time={time}" alt="RRD graph - by day"/><br/>',
-                 '  <img src="rrd.php?name={name}-week.png&time={time}" alt="RRD graph - by week"/><br/>',
-                 '  <img src="rrd.php?name={name}-month.png&time={time}" alt="RRD graph - by month"/><br/>',
-                 '  <img src="rrd.php?name={name}-year.png&time={time}" alt="RRD graph - by year"/>',
+                 '  <img src="rrd.php?name={name}-hour.svg&time={time}" alt="RRD graph - by hour"/><br/>',

```

**Comment by ffloimair** [comment]
I will have a look at it later this evening and will get back to you. Your diff looks different than mine on the js file, meaning I have all replacements as you mentioned in there already (ENV in script, svg instead of png in js file).

**Comment by votdev** [comment]
Sorry, my fault, had some issues while pulling the PR. You can use [Vagrant](https://github.com/openmediavault/openmediavault/tree/master/vagrant) to setup a test system.

File: deb/openmediavault/usr/sbin/omv-mkrrdgraph [hunk 1]
```diff
@@ -48,11 +48,11 @@ def main():
 			'--daemon', 'unix:{}'.format(openmediavault.getenv(
 				'OMV_RRDCACHED_SOCKETFILE', '/var/run/rrdcached.sock')),
 			'--imgformat', openmediavault.getenv(
-				'OMV_COLLECTD_RRDTOOL_GRAPH_IMGFORMAT', 'PNG'),
+				'OMV_COLLECTD_RRDTOOL_GRAPH_IMGFORMAT', 'SVG'),
 			'--width', openmediavault.getenv(
-				'OMV_COLLECTD_RRDTOOL_GRAPH_WIDTH', '400'),
+				'OMV_COLLECTD_RRDTOOL_GRAPH_WIDTH', '600'),
```
Comment (votdev) [inline deb/openmediavault/usr/sbin/omv-mkrrdgraph]: Please use the old size, the big graphs look ugly.

File: deb/openmediavault/usr/sbin/omv-mkrrdgraph [hunk 2]
```diff
@@ -48,11 +48,11 @@ def main():
...
 			'--height', openmediavault.getenv(
-				'OMV_COLLECTD_RRDTOOL_GRAPH_HEIGHT', '120'),
+				'OMV_COLLECTD_RRDTOOL_GRAPH_HEIGHT', '180'),
```
Comment (votdev) [inline deb/openmediavault/usr/sbin/omv-mkrrdgraph]: Please use the old size, the big graphs look ugly.

Comment (ffloimair) [reply deb/openmediavault/usr/sbin/omv-mkrrdgraph]: Done

**Comment by votdev** [comment]
What about the [default image](https://github.com/openmediavault/openmediavault/blob/master/deb/openmediavault/usr/lib/python3/dist-packages/openmediavault/mkrrdgraph.py#L53) that is used if no RRD graph image exists?

**Comment by votdev** [comment]
I still do not like the size of the SVG images. They are rendered to big compared to the PNG version.

auswahl_002

**Comment by ffloimair** [comment]
I played around with the image size as well as the font size in the rrdtool graph. This is the closest to the PNG variant that I came up with.

Concerning the default image: I missed that part.
I see two options here:
1. convert your icon to SVG (do you have a vector graphic as base for this?)
2. make the page render the png only in case of error

I prefer option 1.

**Comment by ffloimair** [comment]
OK, I recreated the error icon as SVG (used your png as template and traced the outlines), as the conversion did not work (wrong colours, odd shapes).

I have now added the requested changes in single commits, but I am planning on squashing this into a single commit as soon as you are OK with it.

**Comment by votdev** [comment]
You can squash it

**Comment by ffloimair** [comment]
Hi!
Can you give me an update of the status. As far as I am concerned I have incorporated all requested changes.

**Comment by ffloimair** [comment]
Well there are some knobs to turn possibly: https://oss.oetiker.ch/rrdtool/doc/rrdgraph_graph.en.html
I can play around a bit and post some suggestions here (pics, but I won't submit any more code).

Interestingly enough it's a question of taste and I actually prefer the SVG variants (but I might be biased due to me working on a 4K display).

**Comment by votdev** [comment]
> I can play around a bit and post some suggestions here (pics, but I won't submit any more code).

Please do so.

> Interestingly enough it's a question of taste and I actually prefer the SVG variants (but I might be biased due to me working on a 4K display).

That's true, but right now the overall look does not match the to rest of the UI because the font/-size. Everything is too big in the graphs. It does not help top reduce the image size.

**Pull request closed without merging**